src/coreclr/jit/inductionvariableopts.cpp

```diff
@@ -2088,6 +2209,25 @@ bool StrengthReductionContext::TryReplaceUsesWithNewPrimaryIV(ArrayStack<CursorI
         m_comp->gtUpdateStmtSideEffects(cursor.Stmt);
     }
 
+    if (m_intermediateIVStores.Height() > 0)
+    {
+        JITDUMP("    Deleting stores of intermediate IVs\n");
+        for (int i = 0; i < m_intermediateIVStores.Height(); i++)
+        {
+            CursorInfo&          cursor = m_intermediateIVStores.BottomRef(i);
+            GenTreeLclVarCommon* store  = cursor.Tree->AsLclVarCommon();
+            JITDUMP("      Replacing [%06u] with a zero constant\n", Compiler::dspTreeID(store->Data()));
+            // We cannot remove these stores entirely as that will break
```

Comment:
We could set the ssa def trees to null, as that is a valid state (see eg #108548), though it might require some tweaking if a lot of code assumes a null def tree means initial value.

Reply:
Or perhaps in some way remove the def from the SSA information, though I suppose that requires updating SSA numbers which requires finding all uses, so not really tractable with our representation.

This seems to work fine for now though, so I'll probably keep it as is, but it definitely falls into our SSA updating woes...